### PR TITLE
(PUP-10382) Ignore packages of which the latest version is not in the index

### PR DESCRIPTION
**Pull request by martijndegouw**
This is a fix for https://tickets.puppetlabs.com/browse/PUP-10382

### CODE REVIEW

**Comment by martijndegouw** [comment]
I think I did something wrong here as well, but I fail to see exactly what's wrong.

**Comment by GabrielNagy** [comment]
Thanks for the contribution! Travis/AppVeyor probably got confused because of the rebase, usually closing and reopening the pull request will fix it.

It would be great if we can have a test for this changed behavior. Looks like [pkgng_spec.rb](https://github.com/puppetlabs/puppet/blob/master/spec/unit/provider/package/pkgng_spec.rb) already has some fixtures to test the `get_latest_version` method, maybe you can make use of them: https://github.com/puppetlabs/puppet/blob/master/spec/unit/provider/package/pkgng_spec.rb#L164

Let me know if you need any help.

File: spec/unit/provider/package/pkgng_spec.rb [hunk 1]
```diff
@@ -169,6 +169,18 @@ def run_in_catalog(*resources)
       bash_comp_latest_version = described_class.get_latest_version('shells/bash-completion', version_list)
       expect(bash_comp_latest_version).to eq('2.1_3')
     end
+
+    it "should return nil when the package is orphaned" do
+      version_list = File.read(my_fixture('pkg.version'))
+      nmap_latest_version = described_class.get_latest_version('sysutils/orphan', version_list)
```
Comment (GabrielNagy) [inline spec/unit/provider/package/pkgng_spec.rb]: nit: this variable still has `nmap` in the title, maybe change to something like `orphaned_latest_version`?

File: spec/unit/provider/package/pkgng_spec.rb [hunk 2]
```diff
@@ -169,6 +169,18 @@ def run_in_catalog(*resources)
...
+      expect(nmap_latest_version).to be_nil
+    end
+
+    it "should return nil when the package is broken" do
+      version_list = File.read(my_fixture('pkg.version'))
+      nmap_latest_version = described_class.get_latest_version('sysutils/broken', version_list)
```
Comment (GabrielNagy) [inline spec/unit/provider/package/pkgng_spec.rb]: nit: this variable still has `nmap` in the title, maybe change to something like `broken_latest_version`?

File: spec/unit/provider/package/pkgng_spec.rb (hunk 1 above)
Comment (martijndegouw) [reply spec/unit/provider/package/pkgng_spec.rb]: the dreaded copy/paste mistake, fixed!

File: spec/unit/provider/package/pkgng_spec.rb (hunk 2 above)
Comment (martijndegouw) [reply spec/unit/provider/package/pkgng_spec.rb]: the dreaded copy/paste mistake, fixed!

**Pull request merged**